src/Core/Collections/Leaderboard/Leaderboard.tsx

```diff
@@ -98,25 +120,57 @@ const Leaderboard = (): ReactElement => {
     }
   };
 
-  const handleSelectTimeRange = (timeRange: LeaderboardTimeRange) => {
-    setLeaderboardTimeRange(timeRange);
-  };
-
   useEffect(() => {
-    handleRequestingLeaderboard();
-  }, [leaderboard, leaderboardTimeRange]);
+    if (Number.isFinite(leaderboardIndex)) {
+      handleRequestingLeaderboard();
+    }
```

Comment:
nit: readability is easier if we just check to see if the index isn't null

```suggestion
  if (leaderboardIndex !== null) {
      handleRequestingLeaderboard();
    }
```